### PR TITLE
feat(UI-Project):Jump to edit release from ProjectDetails

### LINKED ISSUE #372: Allow to jump into "Edit Release" from Project Detail Listings
When being in projects, directly editing a release would be helpful: from a listing of releases, often necessary additions become apparent in clearing status or Linked Releases and therefore having a pencil icon there at the release row and let the user directly jump to edit mode would be helpful.

### PR DESCRIPTION
**Pull request by nutanv1contr**
Added edit button to edit release during project details view and edit. Earlier user was only able to add or delete release , now user will be able to edit the release as well from project details. 

#### How to test:
1. Click on Projects tab and edit project.
2. Go to Linked releases and projects and  click on edit button for the linked releases to edit the release. 

closes #372

### CODE REVIEW

File: frontend/sw360-portlet/src/main/webapp/html/utils/includes/linkedReleasesEdit.jspf [hunk 1]
```diff
@@ -19,7 +19,7 @@
            <col style="width: 20%;" />
            <col style="width: 20%;" />
            <col style="width: 20%;" />
-           <col style="width: 45px;" /> <!-- width of one action (25) with padding of 10 for the column-->
+           <col style="width: 100px;" /> <!-- width of one action (25) with padding of 10 for the column-->
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/utils/includes/linkedReleasesEdit.jspf]: Is the comment still valid?

File: frontend/sw360-portlet/src/main/webapp/html/utils/ajax/linkedReleasesAjax.jsp [hunk 2]
```diff
@@ -52,8 +53,14 @@
                 <sw360:DisplayEnumOptions type="<%=MainlineState.class%>" selected="${releaseLink.mainlineState}"/>
             </select>
         </td>
-        <td class="deletor">
-            <img src="<%=request.getContextPath()%>/images/Trash.png" onclick="deleteReleaseLink('releaseLinkRow${uuid}','<sw360:out value='${releaseLink.longName}' jsQuoting="true"/>')" alt="Delete">
-        </td>
+        <td class="Action">
+          <sw360:DisplayReleaseLink releaseId="${releaseLink.id}"
+               showName="false" page="<%=PortletReleasePage.EDIT%>"><img src='<%=request.getContextPath()%>/images/edit.png'
+               alt='Edit' title='Edit'>&nbsp;&nbsp;
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/utils/ajax/linkedReleasesAjax.jsp]: Why are the `&nbsp;` contained within the `sw360:DisplayReleaseLink`

Comment (maxhbr) [reply frontend/sw360-portlet/src/main/webapp/html/utils/ajax/linkedReleasesAjax.jsp]: Instead of `&nbsp;` for styling, it might be better to use CSS margin

File: frontend/sw360-portlet/src/main/webapp/html/utils/includes/linkedReleasesEdit.jspf (hunk 1 above)
Comment (mcjaeger) [reply frontend/sw360-portlet/src/main/webapp/html/utils/includes/linkedReleasesEdit.jspf]: (from discussion) 70? 95? 120? maybe have a look at other tables

Comment (nutanv1contr) [reply frontend/sw360-portlet/src/main/webapp/html/utils/includes/linkedReleasesEdit.jspf]: Thanks for the suggestion.width: 80px looks feasible and any value less then that is distorting the alignment.So I will be using <col style="width: 80px;" />

File: frontend/sw360-portlet/src/main/webapp/html/utils/ajax/linkedReleasesAjax.jsp (hunk 2 above)
Comment (nutanv1contr) [reply frontend/sw360-portlet/src/main/webapp/html/utils/ajax/linkedReleasesAjax.jsp]: Thanks @maxhbr  for reviewing the changes.I will change  &nbsp; to css style margin

**Comment by nutanv1contr** [comment]
I have pushed the latest changes .Kindly please review the same.

**Comment by mcjaeger** [comment]
sorry, for asking, but the pencil should appear here?

screen shot 2019-02-11 at 21 19 49

(tested on Safari and FF for macosx)

**Comment by mcjaeger** [comment]
(from discussion):
* maybe not in edit mode but view mode (so editing on the project cannot get lost)
* action icon in action column of "Clearing Status"
* On mid-term "linked Releases and Projects" should be merged with "clearing status"

**Comment by nutanv1contr** [comment]
Thanks @mcjaeger  for your valuable feedback. I have made the necessary changes. 
Now on selecting a project from Projects tab , the edit button for edit release will be coming on "Clearing Status" screen.
Previous changes are reverted back.

**Comment by mcjaeger** [comment]
@nutanv1contr I was wondering On which system or on which display do you test? I have a 13" MPB with retina display. attached screenshot when opening the page with Firefox: even at full window width, the icons are not placed horizontally to each other but vertically. Note that horizontal screenshot pixel size is 3360 pixel -> that is not cool as this means the window has an equivalent of ca. 1600 pixels and still the icons are stacked vertically on each other. Tested with Firefox and Safari, both the same output. I recommend the total table layout should be adjusted so that a window with 1200~1400 pixels size will display the icons OK.

**Comment by nutanv1contr** [comment]
Hi @mcjaeger ,
I tested the functionality for 1920*1080 display. As you said after reducing the screen resolution to 1280 * 1024 the buttons are coming vertically. I was able to fix it by increasing table header column style: width allocation. 
The edit button is now coming to the right of "Action" button. I have tested the functionality with 1280*720 resolution display in chrome and Firefox browser.

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf [hunk 3]
```diff
@@ -30,7 +31,7 @@
                 <th></th>
                 <th></th>
                 <th></th>
-                <th></th>
+                <th width="15%"></th>
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf]: This looks like a hack, why is this necessary?

Comment (nutanv1contr) [reply frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf]: Without width allocation the table is distorting and the buttons are coming vertically.On the same table ,there are two other table header for which width allocation is present and also observed the same for few other tables. Please suggest if there is any better approach for this.

Comment (mcjaeger) [reply frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf]: one solution is maybe outlined here (using css classes) https://stackoverflow.com/questions/16100956/prevent-after-element-from-wrapping-to-next-line

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf [hunk 4]
```diff
@@ -66,24 +67,26 @@
...
-            return "<img src='<%=request.getContextPath()%>/images/fossology-logo-24.gif' " +
-                " class='action send-to-fossology' data-release-id='" + releaseId + "'" +
-                "  alt='SelectClearing' title='send to Fossology'>";
-        }
-
         function createClearingTable() {
             var result = [];
 
             <core_rt:forEach items="${releasesAndProjects}" var="releasesClearingStatusData">
+            var sendToFossology = "<img src='<%=request.getContextPath()%>/images/fossology-logo-24.gif' " +
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf]: One minor comment: The resulting JS code looks like:
```
       function createClearingTable() {
￼            var result = [];
            var sendToFossology = [...]
[...]
            var sendToFossology = [...]
[...]
       }
```

I do not know the implications of the multiple `var sendToFossology =` statements in one function, maybe extract add the line `var sendToFossology, editRelease;` to the top of the function and remove the superfluous `var ` statements.

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf [hunk 5]
```diff
@@ -66,24 +67,26 @@
...
         });
 
-        function renderActions(releaseId) {
-            return "<img src='<%=request.getContextPath()%>/images/fossology-logo-24.gif' " +
-                " class='action send-to-fossology' data-release-id='" + releaseId + "'" +
-                "  alt='SelectClearing' title='send to Fossology'>";
-        }
-
         function createClearingTable() {
             var result = [];
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf]:
```suggestion
var sendToFossology, editRelease;
```

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf (hunk 4 above)
Comment (maxhbr) [reply frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf]:
```suggestion
            sendToFossology = "<img src='<%=request.getContextPath()%>/images/fossology-logo-24.gif' " +
```

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf [hunk 6]
```diff
@@ -66,24 +67,26 @@
...
 
             <core_rt:forEach items="${releasesAndProjects}" var="releasesClearingStatusData">
+            var sendToFossology = "<img src='<%=request.getContextPath()%>/images/fossology-logo-24.gif' " +
+                                       " class='action send-to-fossology' data-release-id='" + releaseId + "'" +
+                                       "  alt='SelectClearing' title='send to Fossology'>";
+            var editRelease = "<sw360:DisplayReleaseLink releaseId='${releasesClearingStatusData.release.id}' showName='false' page='<%=PortletReleasePage.EDIT%>'>"+
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf]:
```suggestion
            editRelease = "<sw360:DisplayReleaseLink releaseId='${releasesClearingStatusData.release.id}' showName='false' page='<%=PortletReleasePage.EDIT%>'>"+
```

Comment (nutanv1contr) [reply frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/clearingStatus.jspf]: I have pushed the requested changes.
Thanks

**Comment by smrutis1** [comment]
Tested this in both Chrome and Firefox browser. It is working as expected.
